src/emu/x64primop.h

```diff
@@ -311,7 +311,7 @@ static inline uint8_t shl8(x64emu_t *emu, uint8_t d, uint8_t s)
 		emu->df = d_shl8;
 		emu->op1.u8 = d;
 		emu->op2.u8 = s;
-		emu->res.u8 = d << s;
+		emu->res.u8 = s > 7 ? 0 : d << s;
```

Comment:
that's fine, but I think you need paranthesis, I don't think `:` as more priority than `<<` (at least it's more clear that way).

Reply:
so `(s > 7) ? 0 : (d <<s)` (I also find the space to be useless with the paranthesis, but it's just personal taste)